watchtower/wtdb/client_db.go

```diff
@@ -0,0 +1,902 @@
+package wtdb
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math"
+	"net"
+
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+const (
+	// clientDBName is the filename of client database.
+	clientDBName = "wtclient.db"
```

Comment:
the primary reason for moving both out is that their data is much less critical than the channel info currently stored in channel.db. putting both in channel.db creates more contention and possibility of a failed write producing corruption. the towerdb is also segregated since it could become pretty large, and so that it can be used in a standalone binary and copied over if it is run initially as part of the daemon.

Reply:
The server portion is also designed to be able to eventually run as a standalone binary (no lnd).

Reply:
the server db in theory could also get pretty big, so users may wish to mount on a different volume for regular lnd. adding both into channel.db would cause extra contention and possibility of corrupting more important state like channel info

Reply:
ok, understood for the server db.

but yes, for the client, it seems this is state that could be kept pretty close to the main channel data. it doesn't sound ideal that we create multiple files because we worry about db corruption.

I am wondering, is it possible to then at least use a single db version for both main lnd and wt client dbs? Or will that just create spaghetti without solving anything?

Reply:
>  it doesn't sound ideal that we create multiple files because we worry about db corruption.

to me this is a step in the direction of only needing to update channeldb for critical channel information. ideally, all writes for gossip or anything extraneous that can be recomputed or is otherwise safe to reconstruct should live outside of the channeldb. re: fear of corruption, in theory i agree, but not all hardware is created equal and isolating the two can't make the issue worse than it is today

Reply:
Yes, agree to that. Would you ideally put the graph/gossip data and the wt data in the same db then? So limit it to a critical and a non critical db?

Reply:
In my mind, there'd be at least 3. Critical channel data, non-critical channel data, and gossip data. Rationale behind the graph living in its own db is that it requires the highest throughput, and separating would limit contention on the other databases

Reply:
Ok, makes sense.